mne/minimum_norm/resolution_metrics.py

```diff
@@ -0,0 +1,397 @@
+        print('Not a recognised kind of resolution feature: %s.' % kind)
+
+        return
+
+    if function not in ['psf', 'ctf']:
+        print('Not a recognised resolution function: %s.' % function)
+        return
+
+    if kind == 'localization_error':
+
+        resolution_metric = _localisation_error(resmat, src, function=function,
+                                                metric=metric)
+
+    elif kind == 'spatial_extent':
+
+        resolution_metric = _spatial_extent(resmat, src, function=function,
+                                            metric=metric)
+
+    elif kind == 'relative_amplitude':
+
+        resolution_metric = _relative_amplitude(resmat, src, function=function,
+                                                metric=metric)
+
+    # get vertices from source space
+    vertno_lh = src[0]['vertno']
+    vertno_rh = src[1]['vertno']
+    vertno = [vertno_lh, vertno_rh]
+
+    # Convert array to source estimate
+    resolution_metric = SourceEstimate(resolution_metric, vertno, tmin=0.,
+                                       tstep=1.)
+
+    return resolution_metric
+
+
+def _localisation_error(resmat, src, function, metric):
+    """Compute localisation error metrics for resolution matrix.
+
+    Parameters
+    ----------
+    resmat: array, shape (n_orient*n_locations, n_locations)
+        The resolution matrix.
+        If not a square matrix and if the number of rows is a multiple of
+        number of columns (i.e. n_orient>1), then the Euclidean length per
+        source location is computed (e.g. if inverse operator with free
+        orientations was applied to foward solution with fixed orientations).
+    src: Source Space
+        Source space object from forward or inverse operator.
+    function: str
+        'psf'|'ctf'
+        Whether to compute metrics for columns (point-spread functions, PSFs)
+        or rows (cross-talk functions, CTFs).
+    metric: str
+        What type of localisation error to compute.
+        'peak': Peak localisation error (PLE), Euclidean distance between peak
+                and true source location, in centimeters.
+        'cog': Centre-of-gravity localisation error (CoG), Euclidean distance
+               between CoG and true source location, in centimeters.
+
+    Returns
+    -------
+        locerr: array, shape (n_locations,)
+        Localisation error per location (in cm).
+    """
+    # ensure resolution matrix is square
+    # combine rows (Euclidean length) if necessary
+    resmat = _rectify_resolution_matrix(resmat)
+
+    # locations used in forward and inverse operator
+    locations = _get_src_locations(src)
+
+    # convert to cm (more common)
+    locations = 100. * locations
+
+    # only use absolute values
+    resmat = np.absolute(resmat)
+
+    # The below will operate on columns
+    if function == 'ctf':
+
+        resmat = resmat.T
+
+    # Euclidean distance between true location and maximum
+    if metric.lower() == 'peak':
```

Comment:
```suggestion
    if metric == 'peak':
```

no need to support case errors